lib/controllers/restaurants.js

```diff
@@ -0,0 +1,33 @@
+const { Router } = require('express');
+const authenticate = require('../middleware/authenticate');
+const Restaurant = require('../models/Restaurant');
+
+module.exports = Router()
+  .get('/', async (req, res, next) => {
+    try {
+      const restaurants = await Restaurant.getAll();
+      res.json(restaurants);
+    } catch (e) {
+      next(e);
+    }
+  })
+  .get('/:id', async (req, res, next) => {
+    try {
+      const restaurant = await Restaurant.getByID(req.params.id);
```

Comment:
What happens if `Restaurant.getById(req.params.id)` returns `null`? Would be a good thing to check and possibly set the req.status to 404 and then forward it along to the not-found middleware.